### PR TITLE
eve: Add TestRail upload results step

### PR DESCRIPTION
**Pull request by TeddyAndrieux**
**Component**:

'build'

**Context**: 

TestRail results need to be uploaded from JUnit file generated at build time

**Summary**:

Add a step to upload TestRail results using JUnit file generated at
build time. This upload of JUnit file is done by a tool from test_tools
repository.

---

Fixes: #2336

### CODE REVIEW

File: eve/main.yml [hunk 1]
```diff
@@ -469,6 +469,32 @@ stages:
             --test-suite "$TESTRAIL_SUITE"
             --test-plan "$TESTRAIL_PLAN"
             "$DESCRIPTION_FILE"
+          haltOnFailure: False
+      - ShellCommand:
+          name: Upload TestRail results
+          env:
+            <<: *env_testrail
+            BASE_URL: "%(prop:artifacts_private_url)s/build_status/"
+            REGEXP: "*.xml"
```
Comment (gdemonet) [inline eve/main.yml]: more like a "PATTERN" or "WILDCARD", but whatever :innocent:

**Review by gdemonet — DISMISSED** [review]

File: eve/main.yml (hunk 1 above)
Comment (TeddyAndrieux) [reply eve/main.yml]: Hmm yes PATTERN I like it, I will change it

**Review by gdemonet — APPROVED** [review]

**Pull request merged**